### ISSUE TITLE
SQLite support for Opaleye...?

### ISSUE THREAD

**Issue by mgajda** [tomjaguarpaw/haskell-opaleye#528]
It would be nice to have SQLite backend for opaleye updated.

@tomjaguarpaw So I open ticket just in case.

**Comment by tomjaguarpaw** [comment]
Yes, it would be nice. I don't have any plans to do so but I would be happy to support someone else doing so. Unfortunately it wouldn't be able to have a `Monad` interface because SQLite doesn't support `LATERAL` join.

**Comment by typedrat** [comment]
I'm in a situation where the prospect of this is very interesting to me, so I would be happy to do the work given a certain amount of guidance.

I would assume that at this point trying to use any of `opaleye-sqlite`'s remnants would be pretty unhelpful since it's been decomposing for years now.

**Comment by tomjaguarpaw** [comment]
Cool! I'd be happy to provide guidance where you need it. I would suggest starting by looking at the last commit that touched `opaleye-sqlite` and seeing what the difference was between `opaleye` and `opaleye-sqlite` at that stage. That will tell you the most obvious things that you need to change in the current `opaleye`.
